internal/azurekeyvault/authorizer.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package azurekeyvault provides shared functionality between the
+// signing and secret clients for KeyVault
+package azurekeyvault
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+var keyvaultAuthorizer autorest.Authorizer
+var lock sync.Mutex
+
+// GetKeyVaultAuthorizer prepares a specifc authorizer for keyvault use
+func GetKeyVaultAuthorizer() (autorest.Authorizer, error) {
+	if keyvaultAuthorizer != nil {
```

Comment:
the nil check should go inside the lock - there's a potential race that this gets initialized twice.

Reply:
yep, thanks for pointing that out